clients/src/main/java/org/apache/kafka/clients/Metadata.java

```diff
@@ -181,6 +215,15 @@ public synchronized void update(Cluster cluster, long now) {
     }
 
     /**
+     * If expiry is enabled, remove the unknown topic from the metadata refresh set to
+     * avoid further metadata requests for deleted topics which are not in use.
+     */
+    public synchronized void handleUnknownTopic(String topic) {
```

Comment:
`update()` is invoked only if `cluster.nodes().size() > 0`. I think we want to invoke `handleUnknownTopic` even if cluster is empty. Rather than move the whole logic into `Metadata`, it felt easier to keep two methods that are called from `NetworkClient`. Does that make sense?